Instruments/python/btag_norm_sf.py

```diff
@@ -0,0 +1,380 @@
+import ROOT
+import argparse
+import json
+from numpy import *
+
+
+
+def add_dic(r_factor, r_factor_name, r_factors):
+    print ("r factor related to %s is %.4f " % (r_factor_name ,r_factor))
+    if isnan(r_factor):
+        r_factor = 1
+    r_factors[r_factor_name]=r_factor
+
+def evaluate_r(input_dir,channel, year, unc_source, r_factors):
+    file = input_dir+year+"_"+channel+"_"+unc_source+".root"
+    d = ROOT.RDataFrame(channel, file)
+    not_data = '! is_data'
+    d=d.Filter(not_data)
+    if(unc_source == "Central"):
+        print "\n ******** Central ********* \n"
+        w_central_before = d.Define( "num", "weight").Sum('num')
+        w_central_after = d.Define( "den", "weight*weight_btag_IterativeFit").Sum('den')
+        r_factor=w_central_before.GetValue()/w_central_after.GetValue()
+        add_dic(r_factor, "Central",r_factors)
+
+        w_btag_lf_Up_before = d.Define( "num_btag_lf_Up", "weight*unc_btag_lf_Up").Sum('num_btag_lf_Up')
+        w_btag_lf_Up_after = d.Define( "den_btag_lf_Up", "weight*unc_btag_lf_Up*weight_btag_IterativeFit*unc_btag_lf_Up").Sum('den_btag_lf_Up')
```

Comment:
- weight for denumenator shouldn't be multiplied by unc_btag_lf_Up
- in numerator you should multiply only once by unc_btag_lf_Up
- the code in this way is unreadable: the logic is the same for all unc -> total redesign of the code is needed.